packages/utils/telemetry-utils/src/errorLogging.ts

```diff
@@ -3,102 +3,202 @@
+ * Annotate the given error object, such that it becomes a valid Fluid Error (and is asserted so via a type guard)
+ * If the object is not suitable for annotation (e.g. it's frozen), this will throw.
+ * @param error - The error to annotate by patching in the properties of IFluidErrorBase
+ * @param annotations - Annotations to apply to the normalized error
  */
-type RwLoggingError = LoggingError;
+export function annotateErrorObject(
+    errorObject: unknown,
+    annotations: FluidErrorAnnotations = {},
+): asserts errorObject is IFluidErrorBase {
+    // If errorObject isn't suitable to annotate as IFluidErrorBase, bail.
+    assert(isRegularObject(errorObject) && !Object.isFrozen(errorObject),
+        "Cannot annotate a non-object or frozen error");
+
+    // Patch in any missing properties of IFluidErrorBase
+    const errorTypeIfNone = isErrorLike(errorObject)
+        ? errorObject.name
+        : typeof(errorObject);
+    const fluidError = patchFluidErrorBuilder(
+        errorObject as FluidErrorBuilder,
+        prepareFluidErrorTemplate(errorObject, errorTypeIfNone, annotations.errorCodeIfNone),
+    );
 
-/** Type guard for RwLoggingError.  Not exported. */
-const isRwLoggingError = (x: any): x is RwLoggingError =>
-    typeof x?.addTelemetryProperties === "function" && isILoggingError(x);
+    mixinTelemetryPropsWithFluidError(fluidError, annotations.props);
+}
 
 /**
- * Annotate the given error object with the given logging props
- * @returns The same error object passed in if possible, with telemetry props functionality mixed in
+ * Helper type, not exported.
+ * Makes IFluidErrorBuilder's props mutable and optional for building one up on an existing object
  */
-export function annotateError(
-    error: unknown,
-    props: ITelemetryProperties,
-): ILoggingError {
-    if (isRwLoggingError(error)) {
-        error.addTelemetryProperties(props);
-        return error;
-    }
+type FluidErrorBuilder = {
+    -readonly [P in keyof IFluidErrorBase]?: IFluidErrorBase[P];
+};
+
+/**
+ * Helper type, not exported.
+ * Makes IFluidErrorBuilder's props all type unknown for querying if they're already valid
+ */
+ type FluidErrorProperties = {
+    -readonly [P in keyof IFluidErrorBase]: unknown;
+};
 
-    if (isRegularObject(error)) {
-        // Even though it's not exposed, fully implement IRwLoggingError for subsequent calls to annotateError
-        const loggingError = error as RwLoggingError;
+/** Returns a template IFluidErrorBase with values based on the inputs provided, for use with patchFluidErrorBuilder */
+function prepareFluidErrorTemplate(
+    originalError: unknown,
+    errorTypeIfNone: string,
+    errorCodeIfNone: string | undefined,
+): IFluidErrorBase {
+    const fullErrorTypeIfNone = `none (${errorTypeIfNone})`;
+    const fullErrorCodeIfNone = errorCodeIfNone === undefined
+        ? "none"
+        : `none (${errorCodeIfNone})`;
```

Comment:
I'm not sure if "none " prefix is helping with anything. I.e. given code (assuming it is used only in single place) would either always show up in telemetry with "none" prefix or would not show up at all.

Reply:
I think this is actually quite important to my strategy.  `fluidErrorCode` being a string is the key to distinguish a Fluid Error from an unknown error input, both in the code and in telemetry.  And I'm really looking forward to filtering errors queries down to `Data_fluidErrorCode has "none"` to see what errors are not originating as Fluid Errors.  This represents places to improve our own instrumentation, or "hot spots" where the code or a dependency has some instability.

Reply:
I'd rather it be its own property than rely on string concatenations.

Reply:
do not use the error code field for something like this, it will lead to multiple versions of the runtime have different messages for the same error. if you want to identify this case, just add a new property